Source/Scene/Cesium3DTileset.js

```diff
@@ -0,0 +1,1821 @@
+                attributes++;
+            }
+
+            labelString += '\nFeatures: ' + tile.content.featuresLength;
+            attributes ++;
+        }
+
+        if (tileset.debugShowMemoryUsage) {
+            labelString += '\nTexture Memory: ' + formatMemoryString(tile.content.texturesByteLength);
+            labelString += '\nGeometry Memory: ' + formatMemoryString(tile.content.geometryByteLength);
+            attributes += 2;
+        }
+
+        var newLabel = {
+            text : labelString.substring(1),
+            position : position,
+            font : (19-attributes) + 'px sans-serif',
+            showBackground : true,
+            disableDepthTestDistance : Number.POSITIVE_INFINITY
+        };
+
+        return tileset._tileDebugLabels.add(newLabel);
+    }
+
+    function updateTileDebugLabels(tileset, frameState) {
+        var selectedTiles = tileset._selectedTiles;
+        var length = selectedTiles.length;
+        tileset._tileDebugLabels.removeAll();
+
+        if (tileset.debugPickedTileLabelOnly) {
+            if (defined(tileset.debugPickedTile)) {
+                var position = (defined(tileset.debugPickPosition)) ? tileset.debugPickPosition : computeTileLabelPosition(tileset.debugPickedTile);
+                var label = addTileDebugLabel(tileset.debugPickedTile, tileset, position);
+                label.pixelOffset = new Cartesian2(15, -15); // Offset to avoid picking the label.
+            }
+        } else {
+            for (var i = 0; i < length; ++i) {
+                var tile = selectedTiles[i];
+                addTileDebugLabel(tile, tileset, computeTileLabelPosition(tile));
+            }
+        }
+        tileset._tileDebugLabels.update(frameState);
+    }
+
+    var stencilClearCommand = new ClearCommand({
+        stencil : 0,
+        pass : Pass.CESIUM_3D_TILE
+    });
+
+    function updateTiles(tileset, frameState) {
+        tileset._styleEngine.applyStyle(tileset, frameState);
+
+        var statistics = tileset._statistics;
+        var commandList = frameState.commandList;
+        var numberOfInitialCommands = commandList.length;
+        var selectedTiles = tileset._selectedTiles;
+        var length = selectedTiles.length;
+        var tileVisible = tileset.tileVisible;
+        var i;
+
+        var bivariateVisibilityTest = tileset.skipLevelOfDetail && tileset._hasMixedContent && frameState.context.stencilBuffer && length > 0;
+
+        tileset._backfaceCommands.length = 0;
+
+        if (bivariateVisibilityTest) {
+            commandList.push(stencilClearCommand);
+        }
+
+        var lengthBeforeUpdate = commandList.length;
+        for (i = 0; i < length; ++i) {
+            var tile = selectedTiles[i];
+            // tiles may get unloaded and destroyed between selection and update
+            if (tile.selected) {
+                // Raise visible event before update in case the visible event
```

Comment:
You mean the visible event handlers, right?  This was confusing the first time I read it.